jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpChannelState.java

```diff
@@ -1309,7 +1310,12 @@ public void succeeded()
                 httpChannel.lockedStreamSendCompleted(true);
             }
             if (callback != null)
-                httpChannel._serializedInvoker.run(callback::succeeded);
+            {
+                // Do not use SerializedInvoker here because application code running within
+                // SerializedInvoker could issue blocking write() or close() that would deadlock.
+                // We must execute in case of recursive write within the callback.
+                getRequest().getContext().execute(callback::succeeded);
```

Comment:
We cannot have an execute on the callback of every write.   This is a blocker.  Let's analyse after 12.0.1 and see how we can avoid t his.

Reply:
I think what you need to do is execute the call to the update handler init, so it not within the serialized invoker.